test/studies/champs/sub_test

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Custom sub_test to run CHAMPS testing. Clones, builds CHAMPS and runs testing.
+
+CHAMPS_BRANCH=${CHAMPS_BRANCH:-development}
+
+CHAMPS_PATCH_PATH=$CHAMPS_SUPPORT_PATH/patches
+CHAMPS_GRAPH_PATH=$CHAMPS_SUPPORT_PATH/graph-infra
+CHAMPS_DATA_PATH=$CHAMPS_SUPPORT_PATH/data
+CHAMPS_CFG_PATH=$CHAMPS_SUPPORT_PATH/configs
+
+CWD=$(cd $(dirname ${BASH_SOURCE[0]}) ; pwd)
+source $CWD/functions.bash
+
+subtest_start
+
+export CHAMPS_HOME=$CWD/champs
+rm -rf ${CHAMPS_HOME}
+
+# Clone CHAMPS 
+if ! git clone --depth=1 ${CHAMPS_URL} --branch=${CHAMPS_BRANCH} ; then
+  log_fatal_error "cloning CHAMPS"
+else
+  echo "Cloned CHAMPS successfully"
+fi
+cd ${CHAMPS_HOME}
+
+CHAMPS_VERSION=$(git describe --tags --abbrev=0)
+
+# Apply patches
+for p in $CHAMPS_PATCH_PATH/*patch;
+do
+  apply_patch $p
+done
```

Comment:
Should we be doing something to ensure patches are applied in a particular order? (I can never remember if globs are sorted by default and if they are if it's alphabetical or numeric or what.  I think alphabetical, much means we won't apply in order if we get up to 10 patches.)

Nit, but let's put the `do` on the `for` line to match the `if`/`then` style.

Reply:
Answered the first part, but bash expands globs in alphabetical order. And the patches in the path must be named appropriately for this to work.

I'll address the whitespace issue.